lib/trebuchet.rb

```diff
@@ -117,9 +117,15 @@ def launch(feature, &block)
   end
 
   def launch?(feature)
-    result = !!Feature.find(feature).launch_at?(@current_user, @request)
-    Trebuchet.log(feature, result)
-    return result
+    result = @result_cache[feature]
+
+    if result.nil?
+      result = @result_cache[feature] =
+        !!Feature.find(feature).launch_at?(@current_user, @request)
+      Trebuchet.log(feature, result)
```

Comment:
this means we will bypass logging for a cache hit - is that desirable? is it possible people are expecting the log to mean something else?

Reply:
I made this decision after examining the implementation of `Trebuchet.log`

https://github.com/airbnb/trebuchet/blob/master/lib/trebuchet.rb#L44-L47

The semantics of this method are somewhat suspect to begin with, but, based on how it is set up, repeated calls to log on the same `(feature, result)` are already effectively a no-op.